code/controllers/subsystem/camera.dm

```diff
@@ -0,0 +1,34 @@
+// Subsystem that controls the camera vision tiles
+SUBSYSTEM_DEF(camera)
+	name = "Camera"
+	flags = SS_BACKGROUND | SS_NO_INIT
+	priority = FIRE_PRIORITY_AMBIENCE
+	runlevels = RUNLEVEL_GAME | RUNLEVEL_POSTGAME
+	wait = 2.5 SECONDS
+	offline_implications = "AI camera view won't update. No immediate action is needed."
+	var/list/chunk_queue = list()
+
+/datum/controller/subsystem/camera/fire(resumed)
+	var/static/count = 0
+	var/list/queue = chunk_queue
+	if(count)
+		var/c = count
+		count = 0 //so if we runtime on the Cut, we don't try again.
+		queue.Cut(1, c + 1)
+
+	for(var/datum/camerachunk/chunk as anything in queue)
+		++count
+		chunk.update()
+		if(MC_TICK_CHECK)
+			return
```

Comment:
I feel like its worth just doing `queue -= chunk` and not bothering with the count bollocks, I think the raw remove is gonna be faster overall, especially since its 1 instruction and its not having to replace the var currently on the stack.

Reply:
I was on the fence with this implementation. The list should be small enough at all times to just do a `-= chunk` so I'll change it to that.